server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -265,17 +269,71 @@ protected void doStop() {}
 
     @Override
     protected void doClose() {
-        try {
-            blobStore().close();
-        } catch (Exception t) {
-            logger.warn("cannot close blob store", t);
+        BlobStore store = blobStore;
```

Comment:
I'd expect this to be in a synchronized block

Reply:
I don't see any reason for `sync` here as `blobStore` is volatile. It could be reasonable to add `opened`/`closed` flag to avoid reinit after repo is closed

Reply:
`BlobStoreRepository` is a lifecycle component - no any need for `open`/`close` flag there

Reply:
Lifecycle component is not concurrency-aware, and not all methods that create the blobstore etc. are guarded by "if-not-closed" checks. We need to guard against creating a blobstore / container after the closing has started